src/dom-to-image-more.js

```diff
@@ -870,27 +868,97 @@
+            var value = sourceComputedStyles.getPropertyValue(style);
             var inlineValue = inlineStyles.getPropertyValue(style);
 
             inlineStyles.setProperty(style, root ? 'initial' : 'unset');
-            var initialValue = computedStyles.getPropertyValue(style);
+            var initialValue = sourceComputedStyles.getPropertyValue(style);
 
             if (initialValue !== value) {
-                clonedStyle.setProperty(style, value);
+                targetStyle.setProperty(style, value);
             } else {
-                clonedStyle.removeProperty(style);
+                targetStyle.removeProperty(style);
             }
 
             inlineStyles.setProperty(style, inlineValue);
         }
+    }
 
-        return clonedStyle;
+    function copyUserComputedStyleFast(sourceComputedStyles, parentComputedStyles, targetElement) {
+        var defaultStyle = getDefaultStyle(targetElement.tagName);
+        var targetStyle = targetElement.style;
+
+        util.asArray(sourceComputedStyles).forEach(function(name) {
+            var sourceValue = sourceComputedStyles.getPropertyValue(name);
+            // If the style does not match the default, or it does not match the parent's, set it. We don't know which
+            // styles are inherited from the parent and which aren't, so we have to always check both.
+            if (sourceValue !== defaultStyle[name] ||
+              (parentComputedStyles && sourceValue !== parentComputedStyles.getPropertyValue(name))) {
+                targetStyle.setProperty(name, sourceValue, sourceComputedStyles.getPropertyPriority(name));
+            }
+        });
+    }
+
+    var removeDefaultStylesTimeoutId = null;
+    var sandbox = null;
+    var tagNameDefaultStyles = {};
+
+    function getDefaultStyle(tagName) {
+        if (tagNameDefaultStyles[tagName]) {
+            return tagNameDefaultStyles[tagName];
+        }
+        if (!sandbox) {
+            // Create a hidden sandbox <iframe> element within we can create default HTML elements and query their
+            // computed styles. Elements must be rendered in order to query their computed styles. The <iframe> won't
+            // render at all with `display: none`, so we have to use `visibility: hidden` with `position: fixed`.
+            sandbox = document.createElement('iframe');
+            sandbox.style.visibility = 'hidden';
+            sandbox.style.position = 'fixed';
+            document.body.appendChild(sandbox);
+            // Ensure that the iframe is rendered in standard mode
+            sandbox.contentWindow.document.write('<!DOCTYPE html><meta charset="UTF-8"><title>sandbox</title><body>');
+        }
+        var defaultElement = document.createElement(tagName);
+        sandbox.contentWindow.document.body.appendChild(defaultElement);
+        // Ensure that there is some content, so that properties like margin are applied.
+        defaultElement.textContent = '.';
```

Comment:
Maybe set this to a zero-width non-breaking space instead?

Reply:
As in `defaultElement.textContent = ' ';`?

For block elements like `div`, that results in the defaultElement having a height of `0px` but still having a width. Since we override height to `auto`, it only changes the `block-size`, `persective-origin`, and `transform-origin` being returned on the default style, to a different number. For inline elements like `span`, that results in defaultElement having a width of `0px` but still having a height. In this case it only changes `persective-origin` and `transform-origin` being returned on the default style.

Not sure whether this is desirable or not. Seems to be equivalent. The element being cloned would need to have the CSS property set, and the value would have to match the `defaultElement` computed value, in order for it to cause problems (i.e. omitting the style from the cloned HTML). Very slim chance this would happen.

I'm inclined to stick with some text content (rather than spaces), due to the StackOverflow answer from which I derived this section of code: https://stackoverflow.com/questions/42025329/how-to-get-the-applied-style-from-an-element/42068963#42068963. In his code snippet the author includes
```javascript
// ensure that there is some content, so that e.g. margin is applied
elVanilla.textContent = 'foo';
```

I'm fine with either though. Feel free to change it to a single space if you'd like.

Reply:
A small PITA from using `.`... it's the start of a CSS class in a `<style>` tag, or a *invalid* dot operator in a embedded `<script>` tag. I plan to change it to `;` *as a minimum* in #102.